core/src/hlist.rs

```diff
@@ -272,6 +289,8 @@ macro_rules! gen_inherent_methods {
             where Self: HList,
             {
                 HList::is_empty(self)
+                // this is how it's done at the type-level
+                // <<<Self as HList>::Len as typenum::IsEqual<typenum::U0>>::Output as typenum::Bit>::BOOL
```

Comment:
I've left this in to illustrate that this change doesn't forec the new way of doing things. Things can get pretty gnarly.